File: backend/models/Booking.js
import mongoose from 'mongoose';

const bookingSchema = new mongoose.Schema({
  userId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  stationId: { type: String, required: true }, // Store as string to support both mock string IDs and Mongoose ObjectIds
  slotId: { type: String, required: true },
  scheduledTime: { type: Date, required: true },
  duration_min: { type: Number, required: true },
  status: { type: String, default: 'active', enum: ['active', 'cancelled', 'completed'] },
  vehicleId: { type: String, required: true },
  paymentId: { type: String, required: true }
}, {
  timestamps: true
});

const Booking = mongoose.model('Booking', bookingSchema);
export default Booking;

File: android/frontend/app/src/main/java/com/smartev/assistant/model/Booking.kt
package com.smartev.assistant.model

data class Booking(
    val id: String?,
    val _id: String?,
    val userId: String?,
    val vehicleId: String?,
    val stationId: String?,
    val slotId: String?,
    val date: String?,
    val timeSlot: String?,
    val status: String?, // "pending" | "active" | "completed" | "cancelled"
    val powerDraw_kW: Int?,
    val cost: Double?,
    val createdAt: String?
)

data class BookingRequest(
    val vehicleId: String,
    val stationId: String,
    val slotId: String,
    val date: String,
    val timeSlot: String
)

data class BookingResponse(
    val success: Boolean,
    val bookings: List<Booking>?,
    val booking: Booking?,
    val message: String?
)
